src/systems/filecoin_markets/storage_market/storage_deal/storage_deal_flow.md

```diff
@@ -8,48 +8,41 @@ title: Storage Deal Flow
-- 8. Miners can then recover faults by invoking `StorageMinerActor.RecoverFaults` and have sufficient `StorageDealCollateral` in their available balances. FaultySectors are recommitted and power is only restored at the next PoSt submission. Miners will not be able to invoke `RecoverFaults` when they are in the `Challenged` status.
-- 9. Sectors that are failing for `storagemining.MAX_CONSECUTIVE_FAULTS` consecutive ChainEpochs will be cleared and result in `StoragePowerActor.SlashPledgeCollateral`.
-  - TODO: set `X` parameter
+- 7. Declared faults are penalized to a smaller degree than detected. Miners declare failing sectors by invoking `DeclareTemporaryFaults` with a specified fault duration and associated `TemporaryFaultFee`. Miner will lose power associated with the sector upon `OnSectorTemporaryFaultEffectiveBegin`.
+- 8. The loss of power associated with TemporaryFault will be restored upon `OnSectorTemporaryFaultEffectiveEnd` when the miner is now expected to prove over that sector. Failure to do so will result in unsuccessful ElectionPoSt or unsuccessful SurprisePoSt that leads to detected faults.
 
-# Submit PoSt
 
-On every PoSt Submission, the following steps happen.
+# Detect Faults
 
-- 10. `StorageMinerActor` first verifies the PoSt Submission. If PoSt is done correctly, all `Committed` and `Recovering` sectors will be marked as `Active` and power is credited to these sectors. Payments will be processed for deals that are `Active` by invoking `StorageMarketActor.ProcessStorageDealsPayment`.
-- 11. For all sectors that are off from the `ProvingSet`, these sectors are failing. Increment `FaultCount` on these sectors and if any of these sectors are failing for `MAX_CONSECUTIVE_FAULTS` consecutive `ChainEpoch`, these sectors are terminated and cleared from the network.
-- 13. Process sector expiration. Sectors expire when all deals in that sector have expired. Expired sectors will be cleared and `StorageDealCollateral` for both miners and users returned depending on the state that the sectors are in.
-- 14. Submit `FaultReport` and `PowerReport` to `StoragePowerActor` for slashing and power accounting.
-- 15. Check and ensure that Pledge Collateral is statisfied. TODO: some details are missing here, also related to ProvingPeriod depending on PoSt construction.
-- 16. Update challenge status and add `Committed` sectors received during the challenge to the `Sectors`, `ProvingSet`, and `SectorTable`.
-- 17. All Sectors will be considered in `DetectedFaults` when a miner fail to `SubmitSurprisePoSt` or `SubmitElectionPoSt` (jointly `SubmitPoSt` methods) in a proving period and detected by `onMissedSurprisePoSt` in `CheckSurprisePoStSubmissionHappened` (move to State 18).
```

Comment:
Are these steps no longer needed?

Reply:
They have been covered in other steps. Deal payment, sector expiration and slashing are all done lazily.